watchtower-plugin/src/retrier.rs

```diff
@@ -681,23 +698,24 @@ mod tests {
             .unwrap()
             .is_running());
 
-        // Wait for the remaining time and re-check
-        tokio::time::sleep(Duration::from_secs_f64(MAX_RUN_TIME + HALF_API_DELAY)).await;
-
-        let state = wt_client.lock().unwrap();
-        assert_eq!(
-            state.get_tower_status(&tower_id).unwrap(),
-            TowerStatus::Reachable
-        );
-        assert!(!state.retriers.contains_key(&tower_id));
-        assert!(!state
-            .towers
-            .get(&tower_id)
+        wait_until!(wt_client
+            .lock()
             .unwrap()
-            .pending_appointments
-            .contains(&appointment.locator));
+            .get_retrier_status(&tower_id)
+            .is_none());
 
-        api_mock.assert();
+        {
+            let state = wt_client.lock().unwrap();
+            assert!(state.get_tower_status(&tower_id).unwrap().is_reachable());
+            assert!(!state.retriers.contains_key(&tower_id));
```

Comment:
This should be analogous to `state.get_retrier_status(&tower_id).is_none()`, which we waited on above already.

Reply:
Indeed